### PR TITLE
Implemented `setLookAt` and `lerpLookAt`

### PR DESCRIPTION
**Pull request by FMS-Cat**
## New functions

- `setLookAt( position, target, enableTransition )`
  It moves the camera into `position` , and also make it look at `target` .
- `lerpLookAt( positionA, targetA, positionB, targetB, x, enableTransition )`
  Same as `setLookAt` , but it interpolates between two states.
- `setPosition( position, enableTransition )`
  `setLookAt` without `target` , Stay gazing the current target.
- `setTarget( target, enableTransition )`
  `setLookAt` without `position` , Stay still at the position.
- `getPosition()`
  Return its current position.
- `getTarget()`
  Return its current gazing target.

## Improved procedure

- I added the private function `_sanitizeSphericals()` .
  It clamps `_sphericalEnd.theta` into `0` ~ `TAU` and also make `_spherical` in range between `_sphericalEnd.theta - PI` ~ `_sphericalEnd.theta` + PI,
  It prevents spinning the camera more than `PI` .

## Also

- `target` is now `_target` (marked as private). Don't touch it

### CODE REVIEW

**Comment by yomotsu** [comment]
Thanks! sounds very nice!
I will take a look the changes 👍

File: src/camera-controls.js [hunk 1]
```diff
@@ -553,27 +612,27 @@ export default class CameraControls {
...
 	}
 
-	reset( enableTransition ) {
+	getTarget() {
 
-		this._targetEnd.copy( this._target0 );
-		this._sphericalEnd.setFromVector3( _v3.subVectors( this._position0, this._target0 ) );
-		this._sphericalEnd.theta = this._sphericalEnd.theta % ( 2 * Math.PI );
-		this._spherical.theta    = this._spherical.theta    % ( 2 * Math.PI );
+		return this._targetEnd.clone();
```
Comment (yomotsu) [inline src/camera-controls.js]: How about taking out-vector3 for the argument?

```
getTarget( out ) {

  const _out = out.isVector3 ? out : new THREE.Vector3();
  return _out.copy( this._targetEnd )
```
}

Comment (FMS-Cat) [reply src/camera-controls.js]: AAAA cool!! I'm still pretty bad at such Three.js-friendly API 😅
I'll do this tmrw

**Comment by yomotsu** [comment]
merged! thanks!

**Comment by yomotsu** [comment]
the features are available upon v1.8.0

**Pull request merged**